### PR TITLE
Migrate Delete/Restore/SetDeleteAt methods from ChannelStore to return error interface

### PR DESCRIPTION
**Pull request by rvillablanca**
#### Summary

This is a proposal PR that migrates the method `ChannelStore.Delete`, `ChannelStore.Restore` and `ChannelStore.SetDeleteAt` to return error interface.

Fixes https://github.com/mattermost/mattermost-server/issues/14701

Jira ticket: https://mattermost.atlassian.net/browse/MM-25675

### CODE REVIEW

**Comment by rvillablanca** [comment]
@agnivade reminder to create the required issue :)

**Comment by agnivade** [comment]
@rvillablanca - Some conflicts starting to show now. :)

**Comment by rvillablanca** [comment]
Conflicts were solved, waiting for tests to pass.

File: store/sqlstore/channel_store.go [hunk 1]
```diff
@@ -770,28 +770,28 @@ func (s SqlChannelStore) get(id string, master bool, allowFromCache bool) (*mode
...
+		return errors.Wrap(err, "begin_transaction")
 	}
 	defer finalizeTransaction(transaction)
 
-	appErr := s.setDeleteAtT(transaction, channelId, deleteAt, updateAt)
-	if appErr != nil {
-		return appErr
+	err = s.setDeleteAtT(transaction, channelId, deleteAt, updateAt)
+	if err != nil {
+		return err
```
Comment (agnivade) [inline store/sqlstore/channel_store.go:794]: Let's do another wrap here: `errors.Wrap(err, "setDeleteAt: ")`

Comment (rvillablanca) [reply store/sqlstore/channel_store.go:794]: ok, but colon and space are added by `Error()` so I left them out.

**Comment by agnivade** [comment]
@lindy65 - Some testing around deleting and restoring channels should be sufficient.

**Pull request merged**